services/welding-machine-data-simulator-service/app/services/model_client.py

```
@@ -0,0 +1,176 @@
+                await asyncio.sleep(2 ** attempt)  # 지수 백오프
+
+        print(f"❌ {service_name} 최대 재시도 횟수 초과")
+        return None
+
+    async def predict_welding_data(self, current_data: Dict[str, Any], vibration_data: Dict[str, Any]) -> Dict[str, Any]:
+        """Welding Machine에 전류 + 진동 데이터 예측 요청"""
+        service_name = "welding-machine"
+        service_url = settings.model_services.get(service_name)
+
+        if not service_url:
+            print(f"❌ Welding Machine 서비스 URL을 찾을 수 없습니다.")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)